### PR TITLE
Upload Handler: Fix - Harden SVG upload (ED-2212) 

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: core/files/assets/files-upload-handler.php [hunk 1]
```diff
@@ -89,7 +89,9 @@ public function handle_upload_prefilter( $file ) {
 	}
 
 	protected function is_file_should_handled( $file ) {
-		return $this->is_elementor_media_upload() && $this->get_mime_type() === $file['type'];
+		$ext = pathinfo( $file['name'], PATHINFO_EXTENSION );
+
+		return $this->is_elementor_media_upload() && $this->get_mime_type() === $file['type'] && $this->get_file_type() === $ext;
```
Comment (matipojo) [inline core/files/assets/files-upload-handler.php:94]: Ask @rotemee if we can remove the mime-type check.

**Review by matipojo — CHANGES_REQUESTED** [review]
Nice catch and test!

File: core/files/assets/files-upload-handler.php (hunk 1 above)
Comment (StyleShit) [reply core/files/assets/files-upload-handler.php:94]: Well, It doesn't hurt...

**Review by matipojo — APPROVED** [review]

File: core/files/assets/files-upload-handler.php [hunk 2]
```diff
@@ -45,15 +45,15 @@ final public static function is_enabled() {
...
+		 * @param bool $enabled Whether upload is enabled or not.
 		 */
 		$enabled = apply_filters( 'elementor/files/allow_unfiltered_upload', $enabled );
 
 		return $enabled;
 	}
 
 	final public function support_unfiltered_files_upload( $existing_mimes ) {
-		if ( $this->is_elementor_media_upload() ) {
+		if ( $this->is_elementor_media_upload() && $this->is_enabled() ) {
```
Comment (kobizz) [inline core/files/assets/files-upload-handler.php:56]: Isn't it handled [here](https://github.com/elementor/elementor/blob/master/core/files/assets/files-upload-handler.php#L83)?

**Review by kobizz — CHANGES_REQUESTED** [review]

File: core/files/assets/files-upload-handler.php (hunk 2 above)
Comment (StyleShit) [reply core/files/assets/files-upload-handler.php:56]: Looks like it doesn't 🤷‍♂️ 
I have no idea why, it LGTM.

**Review by kobizz — APPROVED** [review]

**Pull request merged**